frontend/libs/studio-components/src/test-utils/testOnClickCallbackReceivesNativeEvent.ts

```diff
@@ -0,0 +1,28 @@
+import type React from 'react';
+import { fireEvent, type RenderResult } from '@testing-library/react';
```

Comment:
Any reason we use `fireEvent` and not `userEvent`? 😄

Reply:
Yes, it was because of the async vs sync, but i changed to async since userEvent is more "realistic"